contracts/GitcoinPassportDecoder.sol

```diff
@@ -57,58 +84,113 @@ contract GitcoinPassportDecoder is
 
   function _authorizeUpgrade(address) internal override onlyOwner {}
 
+  /**
+   * @dev Gets the EAS contract.
+   */
+  function getEASAddress() public view returns (IEAS) {
+    return eas;
+  }
+
+  /**
+   * @dev Gets providers by version.
+   */
+  function getProviders(uint32 version) public view returns (string[] memory) {
+    return providerVersions[version];
+  }
+
   /**
    * @dev Sets the address of the EAS contract.
    * @param _easContractAddress The address of the EAS contract.
    */
-  function setEASAddress(address _easContractAddress) public onlyOwner {
+  function setEASAddress(address _easContractAddress) external onlyOwner {
+    if (_easContractAddress == address(0)) {
+      revert ZeroValue();
+    }
     eas = IEAS(_easContractAddress);
+    emit EASSet(_easContractAddress);
   }
 
   /**
    * @dev Sets the GitcoinResolver contract.
    * @param _gitcoinResolver The address of the GitcoinResolver contract.
    */
-  function setGitcoinResolver(address _gitcoinResolver) public onlyOwner {
+  function setGitcoinResolver(address _gitcoinResolver) external onlyOwner {
+    if (_gitcoinResolver == address(0)) {
+      revert ZeroValue();
+    }
     gitcoinResolver = IGitcoinResolver(_gitcoinResolver);
+    emit ResolverSet(_gitcoinResolver);
   }
 
   /**
    * @dev Sets the schemaUID for the Passport Attestation.
-   * @param _schemaUID The UID of the schema used to make the user's attestation
+   * @param _schemaUID The UID of the schema used to make the user's passport attestation
+   */
+  function setPassportSchemaUID(bytes32 _schemaUID) public onlyOwner {
+    // TODO: shall we prevent setting a zero address?
```

Comment:
afaik by checking for `bytes32(0)` we are checking for the only possible zero value